src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -266,30 +288,54 @@ void CoreWorkerDirectActorTaskSubmitter::PushActorTask(const ClientQueue &queue,
 
   const auto task_id = task_spec.TaskId();
   const auto actor_id = task_spec.ActorId();
-  const auto counter = task_spec.ActorCounter();
+  const auto actor_counter = task_spec.ActorCounter();
+  const auto task_skipped = task_spec.GetMessage().skip_execution();
   RAY_LOG(DEBUG) << "Pushing task " << task_id << " to actor " << actor_id
-                 << " actor counter " << counter << " seq no "
+                 << " actor counter " << actor_counter << " seq no "
                  << request->sequence_number();
   rpc::Address addr(queue.rpc_client->Addr());
   queue.rpc_client->PushActorTask(
       std::move(request), skip_queue,
-      [this, addr, task_id, actor_id](Status status, const rpc::PushTaskReply &reply) {
+      [this, addr, task_id, actor_id, actor_counter, task_spec, task_skipped](
+          Status status, const rpc::PushTaskReply &reply) {
         bool increment_completed_tasks = true;
+
         if (!status.ok()) {
           bool will_retry = task_finisher_->PendingTaskFailed(
```

Comment:
What happens if `!status.ok() && task_skipped`? I'm not sure if it's okay to mark an already completed task as failed.

Reply:
We wouldn't mark it as failed in this case because we check the task_skipped condition first, right? Maybe this comment was on outdated code